public/app/features/scenes/components/NestedScene.tsx

```diff
@@ -1,27 +1,124 @@
   canCollapse?: boolean;
   canRemove?: boolean;
-  layout: SceneLayout;
   actions?: SceneObject[];
 }
 
 export class NestedScene extends SceneObjectBase<NestedSceneState> {
   static Component = NestedSceneRenderer;
 
+  private sceneCollapser: SceneCollapser;
+  private collapsibleChildren: SceneLayoutChild[] = [];
+
+  constructor(state: NestedSceneState) {
+    super(state);
+
+    this.collapsibleChildren = state.children;
+
+    this.sceneCollapser = new SceneCollapser({
+      canCollapse: Boolean(state.canCollapse),
+      canRemove: Boolean(state.canRemove),
+      isCollapsed: Boolean(state.isCollapsed),
+      onCollapse: this.onToggle,
+      onRemove: this.onRemove,
+    });
+
+    const children = this.buildSceneChildren(Boolean(state.isCollapsed));
+    this.setState({ children });
+  }
+
+  private buildSceneChildren = (isCollapsed: boolean) => {
+    const children = isCollapsed
+      ? []
+      : [
+          new SceneFlexChild({
+            children: this.collapsibleChildren,
+          }),
+        ];
+
+    const actionChildren = this.state.actions ? [...this.state.actions, this.sceneCollapser] : [this.sceneCollapser];
+
+    return [
+      new SceneFlexChild({
+        size: { ySizing: isCollapsed ? 'content' : 'fill' },
+        children: [
+          new SceneFlexLayout({
+            direction: 'column',
+            children: [
+              new SceneFlexChild({
+                size: {
+                  ySizing: 'content',
+                },
+                children: [
+                  new SceneToolbar({
+                    orientation: 'horizontal',
+                    children: [
+                      new SceneFlexLayout({
+                        direction: 'row',
+                        justifyContent: 'space-between',
+                        alignItems: 'center',
+                        children: [
+                          new SceneFlexChild({
+                            size: {
+                              xSizing: 'content',
+                            },
+                            children: [
+                              new SceneCanvasText({
+                                text: this.state.title,
+                              }),
+                            ],
+                          }),
+                          new SceneFlexChild({
+                            size: {
+                              xSizing: 'content',
+                            },
+                            children: [
+                              new SceneFlexLayout({
+                                children: isCollapsed ? [this.sceneCollapser] : actionChildren,
+                              }),
+                            ],
+                          }),
+                        ],
+                      }),
+                    ],
+                  }),
+                ],
+              }),
+              ...children,
+            ],
+          }),
+        ],
+      }),
+    ];
+  };
```

Comment:
Feels like this complicated the NestedScene a lot, not sure it brings many benefits.  

having this row and title & collapse and actions be a scene in itself and stored in children (mixing with original collapsibleChildren), what if from the outside we want to update the original children? that original children array is lost in this (stored in this.collapsibleChildren) but we need to be able to update it from the outside? 

This will also play havoc with the scene editor I am afraid.